test/integration/maker/UtilityMaker.t.sol

```diff
@@ -92,6 +92,29 @@ contract MakerUtilityTest is Test, MakerCommonUtils, ERC20Permit2Utils {
         assertEq(IMakerManager(CDP_MANAGER).count(userDSProxy) - userCdpCountBefore, 1); // cdp count should increase by 1
     }
 
+    function testCannotOpenLockETHAndDrawByInvalidSender(uint256 ethLockAmount, uint256 daiDrawAmount) external {
+        // Calculate minimum collateral amount of ETH and drawing random amount of DAI between minimum and maximum
+        IMakerVat vat = IMakerVat(VAT);
+        bytes32 ilkETH = bytes32(bytes(ETH_JOIN_NAME));
+        (, uint256 rate, uint256 spot, , uint256 dust) = vat.ilks(ilkETH);
+        (uint256 daiDrawMin, uint256 minCollateral) = _getDAIDrawMinAndMinCollateral(spot, dust);
+
+        ethLockAmount = bound(ethLockAmount, minCollateral, 1e22);
+        deal(user, ethLockAmount);
+        uint256 daiDrawMax = _getDAIDrawMaxAmount(ethLockAmount, daiDrawMin, spot, rate);
+        daiDrawAmount = bound(daiDrawAmount, daiDrawMin, daiDrawMax);
+
+        vm.prank(user);
+        vm.expectRevert(IMakerUtility.InvalidAgent.selector);
+        makerUtility.openLockETHAndDraw(
+            ethLockAmount,
+            ETH_JOIN_A,
+            DAI_JOIN,
+            bytes32(bytes(ETH_JOIN_NAME)),
+            daiDrawAmount
+        );
```

Comment:
I think we can simplify the `InvalidAgent` testing by invoking the 'openLockETHAndDraw' function with some default values. For example, `makerUtility.openLockETHAndDraw(address(0),address(0)...)`

Reply:
Fixed. Please help check it again. Thanks.